gptel-openai.el

```diff
@@ -330,28 +341,49 @@ Mutate state INFO with response metadata."
                                 (point) 'gptel nil (point-min))))
           (pcase (get-char-property (point) 'gptel)
             ('response
-             (push (list :role "assistant"
-                         :content (buffer-substring-no-properties (point) prev-pt))
-                   prompts))
+             (when-let ((content (gptel--trim-prefixes
+                                  (buffer-substring-no-properties (point) prev-pt))))
+               (push (list :role "assistant" :content content) prompts)))
+            (`(tool . ,id)
+             (save-excursion
+               (condition-case-unless-debug _err
```

Comment:
Change to `condition-case` as some users run Emacs with `debug-on-error` enabled for other, non-gptel reasons.